src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -106,98 +92,67 @@ public static int BindCoreCLRObject(int jsId, int gcHandle)
                 {
                     obj = h.Target as JSObject;
                 }
-                return obj == null ? 0 : (int)(IntPtr)obj.Handle;
             }
+
+            return obj?.Int32Handle ?? 0;
         }
 
         private static JSObject BindJSType(IntPtr jsIntPtr, int coreType)
         {
-            CoreObject coreObject;
             switch (coreType)
             {
                 case 1:
-                    coreObject = new Array(jsIntPtr);
-                    break;
+                    return new Array(jsIntPtr);
```

Comment:
nit, would be shorter with a pattern matching switch. (assuming I'm calling it the right thing)